cloudify_agent/api/factory.py

```diff
@@ -197,16 +197,21 @@ def save(self, daemon):
             self.storage, '{0}.json'.format(
                 daemon.name)
         )
+
+        if getattr(daemon, 'cluster'):
+            cluster_settings.set_cluster_nodes(daemon.cluster)
+            daemon.cluster = cluster_settings.get_cluster_nodes()
+            active = cluster_settings.get_cluster_active()
+            if active is None:
+                cluster_settings.set_cluster_active(daemon.cluster[0])
```

Comment:
if there's no active, mark the first one as the active; if it's not actually the master, then we'll failover normally